pkg/fleets/fleets.go

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fleets
+
+import (
+	stablev1alpha1 "agones.dev/agones/pkg/apis/stable/v1alpha1"
+	listerv1alpha1 "agones.dev/agones/pkg/client/listers/stable/v1alpha1"
+	"agones.dev/agones/pkg/gameserversets"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// ListGameServerSetsByFleetOwner lists all the GameServerSets for a given
+// Fleet
+func ListGameServerSetsByFleetOwner(gameServerSetLister listerv1alpha1.GameServerSetLister, f *stablev1alpha1.Fleet) ([]*stablev1alpha1.GameServerSet, error) {
+	list, err := gameServerSetLister.List(labels.SelectorFromSet(labels.Set{stablev1alpha1.FleetGameServerSetLabel: f.ObjectMeta.Name}))
```

Comment:
Based on a cursory reading of the code, I think you could even replace the concrete type `*stablev1alpha1.Fleet` with a `metav1.Object` interface. This way, the `ListGameServerSetsByFleetOwner` function might enjoy an even wider usage.

Reply:
If you use the `metav1.Object` interface here, then the name lookup becomes `f.GetName()` as with the lookup below.

Reply:
I've seen that pattern in some object-oriented code, but I'm having a hard time remembering what the pattern might be called. It seems entirely reasonable to me.

Reply:
SGTM! Thanks for the sanity check.

Reply:
I think it's basically a form of [double dispatch](https://en.wikipedia.org/wiki/Double_dispatch).